orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/ServerGroupMetadataTagTask.groovy

```diff
@@ -57,38 +60,14 @@ class ServerGroupMetadataTagTask extends AbstractCloudProviderAwareTask implemen
   }
 
   private List<Map> buildTagOperations(Stage stage) {
-    def tag = [
-      name : "spinnaker:metadata",
-      value: [
-        "stageId"      : stage.id,
-        "executionId"  : stage.execution.id,
-        "executionType": stage.execution.class.simpleName.toLowerCase(),
-        "application"  : stage.execution.application,
-        "user"         : stage.execution.authentication?.user
-      ]
-    ]
-
-    if (stage.execution instanceof Orchestration) {
-      tag.value.description = ((Orchestration) stage.execution).description
-    } else if (stage.execution instanceof Pipeline) {
-      Pipeline pipeline = (Pipeline) stage.execution
-      if (pipeline.name) {
-        tag.value.description = pipeline.name
-      }
-      tag.value.pipelineConfigId = pipeline.pipelineConfigId
-    }
-
-    if (stage.context.reason || stage.context.comments) {
-      tag.value.comments = stage.context.comments ?: stage.context.reason
-    }
-
     def operations = []
     ((StageData) stage.mapTo(StageData)).deployServerGroups.each { String region, Set<String> serverGroups ->
       serverGroups.each { String serverGroup ->
+        Collection<Map<String, Object>> tags = tagGenerators ? tagGenerators.findResults { it.generateTag(stage) } : []
```

Comment:
if there are no tags do we want to do anything?

Reply:
Good point. I will update and actually mark the task as `SKIPPED` in that case.